lib/Doctrine/ORM/Cache/Persister/Entity/NonStrictReadWriteCachedEntityPersister.php

```diff
@@ -87,7 +85,7 @@ public function update($entity)
      */
     private function updateCache($entity, bool $isChanged): bool
     {
-        $class     = $this->metadataFactory->getMetadataFor(get_class($entity));
+        $class     = $this->metadataFactory->getMetadataFor($entity::class);
```

Comment:
Can `$entity` get an `object` type hint? If not, then `$entity::class` shouldn't be used.

Reply:
We shouldn't call `get_class()` either then.

I mean, we could deactivate that rule if we don't feel comfortable about it. But the `::class` syntax should be as (un)safe as a `get_class()` call.